icu4c/source/io/ufile.cpp

```diff
@@ -142,39 +143,56 @@ u_fopen(const char    *filename,
     return result; /* not a file leak */
 }
 
+enum { kFilenameBufCapacity = 256 };
```

Comment:
Shouldn't this be PATH_MAX instead of something arbitrary? I think it's from limits.h. I think putil.cpp uses that value.

Reply:
> Shouldn't this be PATH_MAX instead of something arbitrary?
Well:
1. The old code just had "256" so this is certainly no worse than that.
2. PATH_MAX is not necessarily available on all platforms where this code needs to work. In common/putil.cpp is use is conditionalized on
```
#if !UCONFIG_NO_FILE_IO && ((U_PLATFORM_IS_DARWIN_BASED && (U_PLATFORM != U_PF_IPHONE || defined(U_TIMEZONE))) || U_PLATFORM_IS_LINUX_BASED || U_PLATFORM == U_PF_BSD || U_PLATFORM == U_PF_SOLARIS)
```
or (ICU_DATA_DIR_PREFIX_ENV_VAR is set mainly for Apple platform builds)
```
#if defined(ICU_DATA_DIR_PREFIX_ENV_VAR)
```
3. PATH_MAX is not that useful anyway, since the limits really depend on filesystem implementations; see e.g. https://eklitzke.org/path-max-is-tricky

Reply:
FWIW, I think it might be worth adding a brief comment in the code that this is just an arbitrary sized buffer though, and isn't meant to represent any platform or file-system specific size limitation.

Reply:
You could just check if it's defined. If it's defined, use it.  If it's undefined, default to the current value.

Reply:
Well if we're going to do that, then it feels like maybe it should be conditional on `U_PLATFORM_USES_ONLY_WIN32_API`, and `MAX_PATH` should be used for Windows then...

Reply:
My point was that I _don't_ want to use MAX_PATH even if it _is_ defined, since it could be defined to be a large number (like 4096) that makes no sense for a stack-allocated buffer like this. This buffer size is not meant to be tied to any notion of MAX_PATH; rather, it is only meant to be a not-too-big value (for stack allocation) that will still be larger than almost all filenames/paths in practice; if the name is larger we will dynamically allocate the buffer. We could make it smaller if we wanted to. I can certainly add a comment to this effect.

Reply:
How about this: If MAX_PATH is defined and is a reasonable value for stack allocation (like <= 296) I will use that, otherwise I will use 296 (and explain with comments).

Reply:
FWIW, I'd be happy with just leaving it as an arbitrary fixed-size value, but with a comment to that effect.

However, if @grhoten prefers to have it be tied to `PATH_MAX` / `MAX_PATH`, then I like your suggestion of checking to make sure it's not an overly large value.

Reply:
Meh, I guess the current changes are fine.